File: src/main/cpp/movetype.cpp
/*
 * Copyright (C) 2013-2015 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

#include "movetype.h"

namespace pulse {

bool MoveType::isValid(int type) {
  switch (type) {
    case NORMAL:
    case PAWNDOUBLE:
    case PAWNPROMOTION:
    case ENPASSANT:
    case CASTLING:
      return true;
    default:
      return false;
  }
}

}

File: src/main/java/com/fluxchess/pulse/model/MoveType.java
/*
 * Copyright (C) 2013-2021 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse.model;

public final class MoveType {

	public static final int MASK = 0x7;

	public static final int NORMAL = 0;
	public static final int PAWNDOUBLE = 1;
	public static final int PAWNPROMOTION = 2;
	public static final int ENPASSANT = 3;
	public static final int CASTLING = 4;

	public static final int NOMOVETYPE = 5;

	private MoveType() {
	}
}
